### PR TITLE
Add pre-commit

### PR DESCRIPTION
**Pull request by cclauss**

### CODE REVIEW

File: example/main.py [hunk 1]
```diff
@@ -33,7 +35,7 @@
 engine.stop()
 
 
-"""Saving Voice to a file"""
+# Saving Voice to a file"""
```
Comment (cclauss) [inline example/main.py:38]:
```suggestion
# Saving Voice to a file
```

**Comment by willwade** [comment]
Do you think it would be wise adding a ruff.toml to this repo?  https://docs.astral.sh/ruff/formatter/#__tabbed_1_2

**Comment by cclauss** [comment]
* #339 will provide our way to add `ruff` configuration.

**Comment by willwade** [comment]
ok im going to merge these two ruff ones - then add ruff to pyproject..

**Pull request merged**